### ISSUE TITLE
Proper Handling EEC17

### ISSUE THREAD

**Issue by gumbelmj** [battjt/j1939-84#548]
The transmission rate of EEC17 is 
> Transmitted only after requested.  After request, broadcast rate is 1 s.  Update stopped after key switch cycle.

Given this, the tool believes the PG is only broadcast upon request: 
`Req PGN 64587 from Engine #1 (0) with SPNs 7333`

When the engine is off, the PG is not received as a broadcast message and there's no response to the DS request:
```
09:12:41.6221 DS Request for PGN 64587 to Engine #1 (0) for SPNs 7333
09:12:41.2930 18EA00F9 [3] 4B FC 00 (TX)
Error: Timeout - No Response.
FAIL: 6.1.26.6.a - No DS response for PGN 64587 from Engine #1 (0)

```
When the engine is on the PG is received as a broadcast message:
```
PGN 64587 with Supported SPNs 7333
09:14:29.8840 18FC4B00 [8] FF FF FF FF FF FF 05 02
Found: Electronic Engine Controller 17 from Engine #1 (0): 
  SPN  6893, PEMS Engine Fuel Mass Flow Rate: Not Available
  SPN  6894, Vehicle Fuel Rate: Not Available
  SPN  6895, Engine Exhaust Flow Rate: Not Available
  SPN  7333, Cylinder Fuel Rate: 16.156250 mg/stroke
```

But the tool isn't expecting a value and believes the value should be received on request:
```
09:12:41.6221 DS Request for PGN 64587 to Engine #1 (0) for SPNs 7333
09:12:41.2930 18EA00F9 [3] 4B FC 00 (TX)
Error: Timeout - No Response.
FAIL: 6.1.26.6.a - No DS response for PGN 64587 from Engine #1 (0)

```
Since there's no response to a DS request, the tool tried a global request, in which case the DPF Controller responded with NA:

```
09:12:41.8536 Global Request for PGN 64587 for SPNs 7333
09:12:41.5240 18EAFFF9 [3] 4B FC 00 (TX)
09:12:41.5255 04FC4B55 [8] FF FF FF FF FF FF FF FF
Electronic Engine Controller 17 from DPF Controller (85): 
  SPN  6893, PEMS Engine Fuel Mass Flow Rate: Not Available
  SPN  6894, Vehicle Fuel Rate: Not Available
  SPN  6895, Engine Exhaust Flow Rate: Not Available
  SPN  7333, Cylinder Fuel Rate: Not Available
```

How should the tool handle this case?  Do the SPs on this PG need to have a special case handling? 

There are other PGs with same issues:
- PG 61473 (Engine Speed Sensor Information 1) with SP 4201, 723, 4202, 425, 4204, 4203
- PG 61700 (Engine Speed Sensor Information 2) with SP 7846
- PG 64587 (Electronic Engine Controller 17) with SP 6893, 6894, 6895, 7333

**Comment by gumbelmj** [comment]
From an email:

> Dear Matt,
> 
> Please enter documentation and enhancement tickets for 6.1.26.6.
> 
> Criteria b. And c. Are not well matched to DS queries, and bely my oral direction to ignore broadcast periods for DS requests.  b. And c. Shall be omitted for the current draft and will be reimagined along with improvements for broadcast on request data. This should match the current implementation.
> 
> There are two use cases. 1. PGs that are broadcast at a regular period upon request. 2. PGs that the manufacturer choose to broadcast to satisfy an information architecture need.  The 1st creates conflicts between part 1 and part 2, when the data continues to be broadcast after engine start.  My recollection is that these use cases are not part of the detailed statement of work.
> 
> 6.1.26.6 Fail/warn criteria3:  [Refer to footnotes 23 and 24 for a and b, respectively] 
> a.  Fail if no response/no valid data for any broadcast SPN indicated as supported by the OBD ECU in DM24.  
> b.  Fail if any parameter in a fixed period broadcast message is not broadcast within ± 10% of the specified broadcast period. 
> c.  Fail if any parameter in a variable period broadcast message exceeds 110% of its recommended broadcast period.26 
> d. Fail/warn if any broadcast data is not valid for KOEO conditions as per section A.1Table A-1, Minimum Data Stream Support. 
> e. Warn/info per Table A-1, if an expected SPN from the DM24 support list is provided by a non-OBD ECU. f. Fail/warn per Table A-1, if two or more ECUs provide an SPN listed in Table A-1.  
> 
> Best Wishes,
> 
> Eric
> 
> Sent from my iPad

This is being assigned the 'documentation' label and will be included in Future Enhancements.

**Comment by ericthomasswenson** [comment]
Please combine with #525

**Comment by gumbelmj** [comment]
closing to combine with #525
